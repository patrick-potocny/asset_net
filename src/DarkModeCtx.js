import { createContext } from "react";

export const DarkModeCtx = createContext()