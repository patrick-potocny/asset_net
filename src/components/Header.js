import React, { useEffect, useState } from "react";
import logoLight from "../assets/images/logoLight.svg";
import logoDark from "../assets/images/logoDark.svg";
import moon from "../assets/images/moon.svg";
import sun from "../assets/images/sun.svg";

function Header() {
  const [darkMode, setDarkMode] = useState(true);
  const [icon, setIcon] = useState(moon);
  const [logo, setLogo] = useState(logoLight);

  // Sets mode to light mode if user preffers it
  useEffect(() => {
    if (localStorage.getItem('darkMode') === "false") {
      setDarkMode(false);
    }
  }, []);

  useEffect(() => {
    if (darkMode === true) {
      setIcon(moon);
      setLogo(logoLight);
      document.body.classList.remove("lightMode");
      localStorage.setItem("darkMode", true);
    } else if (darkMode === false) {
      setIcon(sun);
      setLogo(logoDark);
      document.body.classList.add("lightMode");
      localStorage.setItem("darkMode", false);
    }
  }, [darkMode]);

  function handleChange() {
    setDarkMode(!darkMode);
  }

  return (
    <header className="header">
      <img src={logo} className="header__logo" alt="AssetNet" />
      <div className="header__toggle">
        <input
          type="checkbox"
          className="header__checkbox"
          id="checkbox"
          data-testid="dark-mode-toggle"
          checked={darkMode}
          onChange={handleChange}
        />
        <label htmlFor="checkbox" className="header__label">
          <div className="header__ball">
            <img src={icon} alt="mode-icon" />
          </div>
        </label>
      </div>
    </header>
  );
}

export default Header;
