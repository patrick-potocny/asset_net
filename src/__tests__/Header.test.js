import React from "react";
import { render } from "@testing-library/react";
import { DarkModeCtx } from "../DarkModeCtx";
import Header from "../components/Header";
import {screen} from '@testing-library/dom'
import userEvent from "@testing-library/user-event";

describe('Header component', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <DarkModeCtx.Provider value={{ darkMode: false, setDarkMode: jest.fn() }}>
        <Header />
      </DarkModeCtx.Provider>
    );
    
    expect(container).toMatchSnapshot();
  });

  test("call context with true when toggle is clicked", () => {
    const setDarkMode = jest.fn()
  
    render(
      <DarkModeCtx.Provider value={{ darkMode: false, setDarkMode }}>
        <Header />
      </DarkModeCtx.Provider>
    );
  
    const toggle = screen.getByTestId("dark-mode-toggle");
    expect(toggle.checked).toEqual(false);
  
    userEvent.click(toggle)
    expect(setDarkMode).toHaveBeenCalledWith(true)
  });
});