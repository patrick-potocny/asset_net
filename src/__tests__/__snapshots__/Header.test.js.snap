// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <img
      alt="AssetNet"
      class="header__logo"
      src="logoDark.svg"
    />
    <div
      class="header__toggle"
    >
      <input
        class="header__checkbox"
        data-testid="dark-mode-toggle"
        id="checkbox"
        type="checkbox"
      />
      <label
        class="header__label"
        for="checkbox"
      >
        <div
          class="header__ball"
        >
          <img
            alt="mode-icon"
            src="sun.svg"
          />
        </div>
      </label>
    </div>
  </header>
</div>
`;
